Library/BookAdd.aspx

```diff
@@ -21,11 +22,40 @@
             </asp:DropDownList>
             <asp:SqlDataSource ID="Author" runat="server" ConnectionString="<%$ ConnectionStrings:Lib2020ConnectionString %>" SelectCommand="SELECT [ID], [FirstName] + ' ' + [LastName] as Name FROM [Author] ORDER BY [LastName]"></asp:SqlDataSource>
         </div>
+
+        <div>
+            <asp:button id="Save" runat="server" text="Save" onclick="Save_Click" />
+        </div>
+
+        <h3>Copies of Book</h3>
+
+        <fieldset>
+            <div>
+                <asp:DropDownList id="LibraryList" runat="server" AppendDataBoundItems="True" AutoPostBack="True" DataSourceID="Library" DataTextField="LocationName" DataValueField="ID" Visible="false">
+                    <asp:ListItem Value="" Text="Select a Library" />
+                </asp:DropDownList>
+                <asp:SqlDataSource ID="Library" runat="server" ConnectionString="<%$ ConnectionStrings:Lib2020ConnectionString %>" SelectCommand="SELECT [ID], [LocationName] FROM [Library] ORDER BY [LocationName]"></asp:SqlDataSource>
+                <asp:Button ID="AddLibrary" runat="server" text="Add" OnClick="AddLibrary_Click" Visible="false"/>
+                <asp:repeater id="Libraries" runat="server" itemtype="DataRow">
+                    <headertemplate>
+                        <table>
+                            <tr>
+                                <th>Library Name</th>
+                                <th>&nbsp;</th>
+                            </tr>
+                    </headertemplate>
+                    <itemtemplate>
+                        <tr>
+                            <td><%# Item.Field<string>("LocationName") %></td>
+                        </tr>
+                    </itemtemplate>
+                    <footertemplate>
+                        </table>
+                    </footertemplate>
+        </asp:repeater>
+            </div>
+        </fieldset>
```

Comment:
I think we chatted about this in person, but you probably don't want to have the "Book Copies" section on the "Add Book" page. The reason is that you don't have a book ID yet to associated with book copy records/rows. After a user adds a book, you could send them to the Book List page and require them to click "Edit" on a book, or alternatively you could send them to the Edit Book for the book that was just added.